src/fides/api/ops/models/application_config.py

```diff
@@ -82,3 +112,86 @@ def get_api_set_config(cls, db: Session) -> Dict[str, Any]:
         if config_record:
             return config_record.api_set
         return {}
+
+    @classmethod
+    def get_config_set_config(cls, db: Session) -> Dict[str, Any]:
+        """
+        Utility method to get the config_set config dict
+
+        An empty `dict` will be returned if no config settings have been set through config
+        """
+        config_record = db.query(cls).first()
+        if config_record:
+            return config_record.config_set
+        return {}
+
+    @classmethod
+    def set_api_set_config(
+        cls, db: Session, api_set_dict: Dict[str, Any]
+    ) -> ApplicationConfig:
+        """
+        Utility method to set the `api_set` column on the `applicationconfig`
+        db record with the provided dictionary of data.
+
+        Updates are *merged* with any existing `api_set` data in the db.
+        """
+        return cls.create_or_update(db, data={"api_set": api_set_dict})
+
+    @classmethod
+    def set_config_set_config(
+        cls, db: Session, config: FidesConfig
+    ) -> ApplicationConfig:
+        """
+        Utility method to set the `config_set` column on the `applicationconfig`
+        db record by serializing the given `FidesConfig` to a JSON blob.
+        """
+        # need to do this deserialization round-trip to get us a dict
+        # that's serializable as JSON in the db:
+        # pydantic's `.dict()` is NOT serializable as JSON --
+        # see https://github.com/pydantic/pydantic/issues/1409
+        config_dict = loads(config.json())
+        return cls.create_or_update(db, data={"config_set": config_dict})
+
+    @classmethod
+    def get_resolved_config_property(
+        cls, db: Session, config_property: str, default_value: Any = None
+    ) -> Optional[Any]:
+        """
+        Gets the 'resolved' config property based on api-set and config-set configs.
+        `config_property` is a dot-separated path to the config property,
+        e.g. `notifications.notification_service_type`.
+
+        Api-set values get priority over config-set, in case of conflict.
+        """
+        config_record = db.query(cls).first()
+        if config_record:
+            api_prop = _get_property(config_record.api_set, config_property)
+            if api_prop is None:
+                logger.info(f"No API-set {config_property} property found")
+                return _get_property(
+                    config_record.config_set, config_property, default_value
+                )
+            return api_prop
+        logger.warning("No config record found!")
+        return default_value
+
+
+def _get_property(
+    properties: Any, property_path: str = None, default_value: Any = None
+) -> Any:
```

Comment:
According to the instance check `properties` has to be a `dict`. Can we change the type from `Any` to `dict`? Also from my other comment, can we remove `default_value`?. If the answer to both of these is yes, and if the `properties` dict is a hashable dict we could add an LRU cache here so it only needs to do the recursive lookup one time.

```py
from functools import lru_cache

@lru_cache(maxsize=1028)  # I'm not sure what a good maxsize would be here. depends on how many different properties there could be.
def _get_property(properties: dict, property_path: str | None = None) -> dict:
    ...
```

Either way `property_path` should be `property_path: str | None = None` instead of just `str`

Reply:
Based on your above descriptions it made me realize a cache won't work here because the value of the dictionary could be an object which is fine. So you can ignore the `lru_cache` suggestion.

So now the remaining questions are should `properties` be `dict` (with more specific types included if possible) and is `Any` correct for `default_values`?

Then `property_path` should be `property_path: str | None = None`

Reply:
agreed that we could make `property_path: str | None = None`, but `properties` doesn't actually have to be a `dict`. when it gets to the end of the recursion, that will be the returned value, which isn't necessarily a `dict`. the instance check is only if `property_path` isn't `None`. 

i agree there may be ways to optimize - with an lru_cache or otherwise, both to make the code more efficient and more readable/reusable. but maybe we could take some of those up in a follow-up?  i'm sure we'll be taking a look again soon enough at this class to expand its use :) 

unless you think we may get unexpected behavior with how it is now.

Reply:
Got it, that makes sense. I agree, lets go with it as is now and can optimize later if needed.